packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -571,66 +708,61 @@ export class ResourceService {
       }
     }
 
-    // 3.delete entity from source service by flag
-    for (const entity of entitiesToCopy) {
-      if (entity.shouldDeleteFromSource) {
-        await this.entityService.deleteEntityFromSource(
-          { where: { id: entity.id } },
-          user
-        );
-      }
+    // 3.delete entities from source services
+    for (const entity of entitiesWithFields) {
+      await this.entityService.deleteEntityFromSource(
+        { where: { id: entity.id } },
+        user
+      );
     }
 
-    return resource;
+    return resources;
   }
 
-  async createResourceCopiedEntities(
-    entitiesToCopy: Entity[],
+  async createResourceCopiedEntity(
+    entityToCopy: Entity,
     targetResourceId: string,
     user: User
-  ): Promise<Entity[]> {
-    const copiedEntities: Entity[] = [];
+  ): Promise<Entity> {
+    const {
+      name,
+      displayName,
+      pluralDisplayName,
+      description,
+      customAttributes,
+    } = entityToCopy;
 
-    for (const entity of entitiesToCopy) {
-      const {
-        name,
-        displayName,
-        pluralDisplayName,
-        description,
-        customAttributes,
-      } = entity;
+    let copiedEntity: Entity;
 
-      try {
-        const copiedEntity = await this.entityService.createOneEntity(
-          {
-            data: {
-              resource: {
-                connect: {
-                  id: targetResourceId,
-                },
+    try {
+      copiedEntity = await this.entityService.createOneEntity(
+        {
+          data: {
+            resource: {
+              connect: {
+                id: targetResourceId,
               },
-              name,
-              displayName,
-              pluralDisplayName,
-              description,
-              customAttributes,
             },
+            name,
+            displayName,
+            pluralDisplayName,
+            description,
+            customAttributes,
           },
-          user,
-          false,
-          false,
-          false
-        );
-        copiedEntities.push(copiedEntity);
-      } catch (error) {
-        this.logger.error(error.message, error, { entity: entity.name });
-        throw new Error(
-          `Failed to create entity "${entity.name}" due to ${error.message}`
-        );
-      }
+        },
+        user,
+        false,
+        false,
+        false
+      );
+    } catch (error) {
+      this.logger.error(error.message, error, { entity: entityToCopy.name });
+      throw new Error(
+        `Failed to create entity "${entityToCopy.name}" due to ${error.message}`
+      );
     }
 
-    return copiedEntities;
+    return copiedEntity;
   }
 
   async createCopiedEntityIdField(copiedEntityId: string): Promise<void> {
```

Comment:
the name "copiedEntities" does not mean anything to anyone who is not familiar with the context